#8688gv6gb: Community key settings

arojas1 opened the pull request:
Added API Key manager to community settings. Branching off of #216 - researcher api key settings added.

arojas1 commented:
Task linked: [CU-8688gv6gb Communities: API Key Account Settings UI](https://app.clickup.com/t/8688gv6gb)

lc-hd reviewed (changes requested):
When a user has no api-keys remaining, this is what I see.  ✔️

![image](https://github.com/localcontexts/localcontextshub/assets/145378945/83e79375-880c-400e-9cd5-a7db512d4636)

--------------------


When a user has api-keys remaining, this is what I see. ✔️

![image](https://github.com/localcontexts/localcontextshub/assets/145378945/eddb6c60-6490-4a38-b28c-d61149ff092d)

-----

I was able to generate an API request via the UI. This is what I saw: ✔️

![image](https://github.com/localcontexts/localcontextshub/assets/145378945/7146c235-9fe1-4e10-9ff4-a8ba4002dd0c)

-----

I was able to cause an error in the api key generation due to form validation which is great. But the error message didn't match the failure reason. Could you create separate error messages when the account is not approved and then another when the form fails validation?


https://github.com/localcontexts/localcontextshub/assets/145378945/c24c2ca9-2f30-4144-ab03-05016e78bcf3

---------------------------------

Deleting the api key works. Could you create a success or info message when that deletion occurs? I think that would add to the user experience.


https://github.com/localcontexts/localcontextshub/assets/145378945/0aa0536e-f558-4fda-9639-36847ec220a2

arojas1 commented:
> I was able to cause an error in the api key generation due to form validation which is great. But the error message didn't match the failure reason. Could you create separate error messages when the account is not approved and then another when the form fails validation?

Fixed this for all accounts to clean data being passed via the form. Fixed to show proper error message.

> Deleting the api key works. Could you create a success or info message when that deletion occurs? I think that would add to the user experience.

Added!
